test/integration/cmn/client/user.go

```diff
@@ -1,69 +1,67 @@
 package client
 
 import (
+	"crypto/ecdsa"
 	"crypto/ed25519"
 	"crypto/rand"
 	"errors"
-	"fmt"
+	"math/big"
 	"strings"
 
+	pbfound "github.com/anoideaopen/foundation/proto"
 	"github.com/btcsuite/btcutil/base58"
+	eth "github.com/ethereum/go-ethereum/crypto"
 	"golang.org/x/crypto/sha3"
 )
 
 type UserFoundation struct {
-	PrivateKey         ed25519.PrivateKey
-	PublicKey          ed25519.PublicKey
+	PrivateKeyBytes    []byte
+	PublicKeyBytes     []byte
+	PublicKeyType      string
 	PublicKeyBase58    string
 	AddressBase58Check string
 	UserID             string
 }
 
-func NewUserFoundation() *UserFoundation {
-	publicKey, privateKey, err := ed25519.GenerateKey(rand.Reader)
-	if err != nil {
-		return &UserFoundation{}
-	}
-	publicKeyBase58 := base58.Encode(publicKey)
-	hash := sha3.Sum256(publicKey)
-	addressBase58Check := base58.CheckEncode(hash[1:], hash[0])
+func NewUserFoundation(keyType string) *UserFoundation {
+	var (
+		privateKeyBytes []byte
+		publicKeyBytes  []byte
+		err             error
+	)
 
-	return &UserFoundation{
-		PrivateKey:         privateKey,
-		PublicKey:          publicKey,
-		PublicKeyBase58:    publicKeyBase58,
-		AddressBase58Check: addressBase58Check,
-		UserID:             "testuser",
-	}
-}
+	switch keyType {
+	case pbfound.KeyType_ed25519.String():
+		publicKeyBytes, privateKeyBytes, err = ed25519.GenerateKey(rand.Reader)
+		if err != nil {
+			return &UserFoundation{}
+		}
+
+	case pbfound.KeyType_secp256k1.String():
+		var privateKey *ecdsa.PrivateKey
+		privateKey, err = ecdsa.GenerateKey(eth.S256(), rand.Reader)
+		if err != nil {
+			return &UserFoundation{}
+		}
+		privateKeyBytes = privateKey.D.Bytes()
+		publicKeyBytes = append(privateKey.X.Bytes(), privateKey.Y.Bytes()...)
 
-func UserFoundationFromPrivateKey(privateKey ed25519.PrivateKey) (*UserFoundation, error) {
```

Comment:
Why did you remove the UserFoundationFromPrivateKey, UserFoundationFromBase58CheckPrivateKey, MultiSig (MultiSign) functions? 
They should be left, at least with a check for type ed25519. 
It seems to me that MultiSign is needed for any type of key.

Reply:
Those functions were not used in the code. I can bring them back

Reply:
it's better to return